src/main/scala/xiangshan/frontend/icache/ICache.scala

```diff
@@ -57,9 +58,10 @@ case class ICacheParameters(
     ICacheDataBanks:     Int = 8,
     ICacheDataSRAMWidth: Int = 66,
     // TODO: hard code, need delete
-    partWayNum: Int = 4,
-    nMMIOs:     Int = 1,
-    blockBytes: Int = 64
+    partWayNum:       Int = 4,
+    nMMIOs:           Int = 1,
+    blockBytes:       Int = 64,
+    cacheCtrlAddress: AddressSet = AddressSet(0x38022080, 0x7f)
```

Comment:
Why default to a magic number? I think DCacheParameters is more reasonable.

Reply:
I don't want this to be optional, currently it is impossible to completely disable ICache ECC so it must have an address.

Reply:
I've got an idea, I can implement a `DummyCtrlUnit`, instantiate it instead of the real `CtrlUnit` when the configuration has no address. In which we hardwire the control signals to the default (enable=1, inject=0), so no other logic needs to be changed. What do you think.

Reply:
This sounds good to me. But how does DCache implement `None`?

Reply:
DCache use `CtrlUnit` for error injection/counting, not for controlling ECC enable. So when it does not get an address, it will only disable the injection/counting ability. As I understand it, DCache uses the compile-time `enableTag/DataEcc` parameter to switch on/off the entire ECC ability, and doesn't allow software to dynamically control the enablement. ICache does the opposite, it doesn't implement the compile-time switch, but does allow software to control it via `ICacheCtrlUnit`.

Maybe we can implement a compile-time switch, but not in this PR.

Reply:
It's just a copy-and-paste mistake. I'll default `cacheCtrlAddress` to `None` in ICache.scala and set it in Configs.scala / Parameters.scala, as DCache does.